### ISSUE TITLE
FunctionDeclarationArgumentSpacingSniff uses strlen on content not spaces

### ISSUE THREAD

**Issue by webimpress** [squizlabs/PHP_CodeSniffer#2285]
The `FunctionDeclarationArgumentSpacingSniff` uses `strlen` on token content, so technically it could be `\t` not the space, and the error is not going to be reported. But configuration says about number of spaces and fixer uses spaces.

**Comment by jrfnl** [comment]
_Great minds think alike_ I actually created a PR-branch for all instances where token `length` should be used instead of `strlen()` this weekend, but am having issues with my internet connection. Hoping the connection will be stable enough tomorrow to pull it ;-)

**Comment by gsherwood** [comment]
Do the `strlen` changes actually fix this? Previously, the sniff was doing a strlen on the `content` index, which should already have been converted from tabs to spaces. Using the `length` index should only save time - not actually give you a different result.

**Comment by gsherwood** [comment]
I tested this by using a single tab between function arguments in the following code:
```php
<?php
function foo($a,	$b,	$c){}
```
When running PHPCS with no `tab-width` setting, you get no errors. When you set the `tab-width` to 4 spaces, you get ` Expected 1 space between comma and argument "$b"; 4 found`. The same error is reported before and after PR #2290, so I don't think it has any effect and everything seems to be working as normal.

What are the steps to replicate this bug?

**Comment by webimpress** [comment]
@gsherwood Maybe I understand wrong how the sniff should work and how we should treat `tab-width` option. In description of properties we have spaces and fixer uses spaces. So example with tabs shouldn't pass (when we are not setting `tab-width`).

I have tried similar code:
```php
<?php // phpcs:set Squiz.Functions.FunctionDeclarationArgumentSpacing equalsSpacing 1

function a($a	=	1,	$b){}
```

(one tab before and after `=` and one tab after `,`).

and run the sniff on it:
```console
$ bin/phpcs tabs.php --sniffs=Squiz.Functions.FunctionDeclarationArgumentSpacing --tab-width=4 -s
```

and here is the result:

```
FOUND 3 ERRORS AFFECTING 1 LINE
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 3 | ERROR | [x] Incorrect spacing between argument "$a" and equals sign; expected 1 but found 3 (Squiz.Functions.FunctionDeclarationArgumentSpacing.SpaceBeforeEquals)
 3 | ERROR | [x] Incorrect spacing between default value and equals sign for argument "$a"; expected 1 but found 3
   |       |     (Squiz.Functions.FunctionDeclarationArgumentSpacing.SpaceAfterDefault)
 3 | ERROR | [x] Expected 1 space between comma and argument "$b"; 2 found (Squiz.Functions.FunctionDeclarationArgumentSpacing.SpacingBeforeArg)
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 3 MARKED SNIFF VIOLATIONS AUTOMATICALLY
```

and what is wrong:
- `expected 1 but found 3` x2
- `Expected 1 .... 2 found`

everywhere we have one `\t`.

**Comment by gsherwood** [comment]
When you use `tab-width` PHPCS converts tabs into the number of spaces they should be (anywhere from `1` to `$tabWidth`, depending on the tab stop). Pretty much all the sniffs only work with spaces, so they report and fix in terms of spaces. Another sniff is responsible for then converting spaces back into tabs during fixing (the DisallowSpaceIndent sniff).

This sniff looks to be working exactly as it was designed. So maybe you're really asking for a tab-aware re-design of the sniff rather than having anything to do with strlen usage?
